engine/table/src/main/java/io/deephaven/engine/table/impl/select/python/ArgumentsChunked.java

```diff
@@ -3,22 +3,54 @@
  */
 package io.deephaven.engine.table.impl.select.python;
 
+import io.deephaven.base.Pair;
 import io.deephaven.chunk.attributes.Any;
 import io.deephaven.chunk.*;
+import io.deephaven.engine.util.PythonScopeJpyImpl;
+import io.deephaven.engine.util.PythonScopeJpyImpl.CallableWrapper;
 import io.deephaven.util.PrimitiveArrayType;
 
+import java.util.ArrayList;
 import java.util.Objects;
 
 class ArgumentsChunked {
-    static ArgumentsChunked buildArguments(Chunk<?>[] __sources) {
+    // static ArgumentsChunked buildArguments(Chunk<?>[] __sources) {
+    // final Class<?>[] paramTypes = new Class[__sources.length];
+    // final Object[] params = new Object[__sources.length];
+    // for (int i = 0; i < __sources.length; i++) {
+    // final ChunkToArray<?> cta = __sources[i].walk(new ChunkToArray<>());
+    // paramTypes[i] = Objects.requireNonNull(cta.getArrayType());
+    // params[i] = Objects.requireNonNull(cta.getArray());
+    // }
+    // return new ArgumentsChunked(paramTypes, params);
+    // }
+
+    static ArgumentsChunked buildArguments(Chunk<?>[] __sources, CallableWrapper callableWrapper) {
         final Class<?>[] paramTypes = new Class[__sources.length];
         final Object[] params = new Object[__sources.length];
         for (int i = 0; i < __sources.length; i++) {
             final ChunkToArray<?> cta = __sources[i].walk(new ChunkToArray<>());
             paramTypes[i] = Objects.requireNonNull(cta.getArrayType());
             params[i] = Objects.requireNonNull(cta.getArray());
         }
-        return new ArgumentsChunked(paramTypes, params);
+
+        ArrayList<Pair<?, ?>> args = callableWrapper.getArgs();
+        final Object[] chunkedArgs = new Object[args.size()];
+        Class<?>[] argTypes = callableWrapper.getArgTypes();
+        Class<?>[] chunkedArgTypes = new Class[argTypes.length];
+
+        for (int i = 0; i < args.size(); i++) {
+            Pair<?, ?> arg = args.get(i);
+            if (arg.getFirst() != null) {
+                int colIdx = (int) arg.getFirst();
+                chunkedArgs[i] = params[colIdx];
+                chunkedArgTypes[i] = paramTypes[colIdx];
+            } else {
+                chunkedArgs[i] = arg.getSecond();
+                chunkedArgTypes[i] = argTypes[i];
+            }
+        }
+        return new ArgumentsChunked(chunkedArgTypes, chunkedArgs);
```

Comment:
I'm not sure that it needs to be part of *this* PR, but we should be doing something about this code.
1. We don't want to allocate a new array for every input argument for every single chunked evaluation. We should be caching those arrays, and really we should be caching the `ArgumentsChunked` instance.
2. Can we avoid copying from our input chunks to our input arrays? I think we can, by using chunks that we own the arrays to, or similar strategies.